redis/src/cluster_routing.rs

```diff
@@ -397,7 +397,12 @@ fn base_routing(cmd: &[u8]) -> RouteBy {
         | b"XGROUP SETID"
         | b"XINFO CONSUMERS"
         | b"XINFO GROUPS"
-        | b"XINFO STREAM" => RouteBy::SecondArg,
+        | b"XINFO STREAM"
+        | b"ZDIFF"
+        | b"ZINTER"
+        | b"ZINTERCARD"
+        | b"ZMPOP"
+        | b"ZUNION" => RouteBy::SecondArg,
```

Comment:
```suggestion
        | b"LMPOP"
        | b"BLMPOP"
        | b"ZUNION" => RouteBy::SecondArg,
```
and also Yurri's additions
please check if `FCALL` should be there too

Reply:
I added the commands Yury requested, LMPOP and ZUNION are now routed to SecondArgAfterKeyCount and BLMPOP is mapped to ThirdArgAfterKeyCount. FCALL is under the Uncategorized section of the script since the key argument is optional. For now I've left it out of the cluster_routing.rs file

Reply:
From the discussion below I've realized cluster_routing can handle optional keys if there is a numkeys argument before the optional key, so I've updated the script to categorize these cases and moved FCALL to ThirdArgAfterKeyCount